### ISSUE TITLE
[docs infra] Does the linter run in the CI?

### ISSUE THREAD

**Issue by emmyscode** [ray-project/ray#40244]
### Description

# Issue
We currently ask users to run the script `format.sh` to format their changes before contributing changes to `ray`. My understanding is that the CI has a job that runs the same script to make sure the changes are formatted properly and will fail if they are not. 

However, after looking at others' pull requests, I can't seem to find the CI job in charge of running the linter. Does anyone know if [`format.sh`](https://github.com/ray-project/ray/blob/dac65160df288266d9d4da91e63b4305353d6ade/ci/lint/format.sh#L4) is being run in the CI, and if so, which job it is part of?

## Background

I tried to follow the instructions for setting up the linter from the [Contributing to the Ray Documentation](https://docs.ray.io/en/latest/ray-contribute/docs.html#install-dependencies) page, but ran into dependency issues, similar to https://github.com/ray-project/ray/issues/39962.

I upgraded `flake8` and `flake8-quotes`, which allowed the linter to run, but it will throw the following error:
```
WARNING: Ray uses flake8 3.9.1, You currently are using 6.1.0. This might generate different results.
WARNING: Ray uses flake8 with flake8_quotes. Might error without it. Install with: pip install flake8-quotes
```

This is because the [`format.sh`](https://github.com/ray-project/ray/blob/dac65160df288266d9d4da91e63b4305353d6ade/ci/lint/format.sh#L4) file expects `3.9.1`.

It might be useful to update the [Contributing to the Ray Documentation](https://docs.ray.io/en/latest/ray-contribute/docs.html#install-dependencies) page to advise people to upgrade `flake8` and `flake8-quotes` manually. This will throw warnings when running the linter locally, which is not ideal but palatable. However, if [`format.sh`](https://github.com/ray-project/ray/blob/dac65160df288266d9d4da91e63b4305353d6ade/ci/lint/format.sh#L4) is run in the CI, then there could be discrepancies between the way formatting changes locally versus what is expected in the build, causing the test to fail. This would lead to confusion by users because the instructions in the docs would cause an error in their pull request. 

If `format.sh` isn't run in the CI, then we can potentially explore changing the version of `flake8` required in the script, avoiding the need to change the instructions in the docs.

**Comment by matthewdeng** [comment]
Yeah it is run in `premerge` job - example [here](https://buildkite.com/ray-project/premerge/builds/8100#018b1b50-80dd-41be-afd6-4e6380e5e8d3).

**Comment by dstrodtman** [comment]
Closing as obsolete. The docs build or infra concern referenced here predates the current toolchain, so the specific complaint no longer applies. If a similar issue exists against the current build, please open a fresh report.

Context: docs-team triage of the OSS docs issue backlog.
